types/index.d.ts

```diff
@@ -0,0 +1,74 @@
+// TypeScript Version: 4.0
+
+import Vue, {ComponentOptions} from 'vue'
+import {ThisTypedMountOptions, VueClass} from '@vue/test-utils'
+import {Store, StoreOptions} from 'vuex'
+import Router, {RouteConfig} from 'vue-router'
+// eslint-disable-next-line import/no-extraneous-dependencies
+import {OptionsReceived as PrettyFormatOptions} from 'pretty-format'
+import {queries, EventType, BoundFunctions} from '@testing-library/dom'
+
+// NOTE: fireEvent is overridden below
+export * from '@testing-library/dom'
+
+export interface RenderResult extends BoundFunctions<typeof queries> {
+  container: HTMLElement
+  baseElement: HTMLElement
+  debug: (
+    baseElement?:
+      | HTMLElement
+      | DocumentFragment
+      | Array<HTMLElement | DocumentFragment>,
+    maxLength?: number,
+    options?: PrettyFormatOptions,
+  ) => void
+  unmount(): void
+  isUnmounted(): boolean
+  html(): string
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  emitted(): {[name: string]: any[][]}
+  updateProps(props: object): Promise<void>
+}
+
+export interface RenderOptions<V extends Vue, S = {}>
+  // The props and store options special-cased by Vue Testing Library and NOT passed to mount().
+  extends Omit<ThisTypedMountOptions<V>, 'store' | 'props'> {
+  props?: object
+  store?: StoreOptions<S>
+  routes?: RouteConfig[]
+  container?: HTMLElement
+  baseElement?: HTMLElement
+}
+
+export type ConfigurationCallback<V extends Vue> = (
+  localVue: typeof Vue,
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  store: Store<any>,
+  router: Router,
+) => Partial<ThisTypedMountOptions<V>> | void
```

Comment:
This line productes the following warning with latest config:

```
void is only valid as a return type or generic type variable
@typescript-eslint/no-invalid-void-type
```

not entirely sure I should be worries about this?